js/kucoin.js

```diff
@@ -1102,6 +1102,34 @@ module.exports = class kucoin extends Exchange {
     }
 
     async createOrder (symbol, type, side, amount, price = undefined, params = {}) {
+        /**
+         * @method
+         * @name kucoin#createOrder
+         * @description Create an order on the exchange
+         * @param {str} symbol Unified CCXT market symbol
+         * @param {str} type "limit" or "market"
+         * @param {str} side "buy" or "sell"
+         * @param {float} amount the amount of currency to trade
+         * @param {float} price *ignored in "market" orders* the price at which the order is to be fullfilled at in units of the quote currency
+         * @param {dict} params  Extra parameters specific to the exchange API endpoint
+         * @param {float} params.leverage Leverage size of the order
```

Comment:
We should not use the js-specific syntax in the docs, so we must replace `params.leverage` with `params['leverage']` and all other similar cases everywhere to keep it uniform with the other languages.

Tbh, I don't really like the idea of moving these params to the docstrings, cause that makes the code less readable.

Reply:
Leave this as is and I'll add a transpilation. This is more readable in my opinion, and it's available to the user, without this, the only way the user can really know how to set leverage on kucoin is by checking kucoins documentation, or searching CCXT issues or examples.

How is this appearing in you editor? The highlighting alone makes this much more readable for me

![](https://i.imgur.com/4N10kj6.png)

Reply:
Let's still do _replace params.leverage with params['leverage']_ for it to work with all languages.

Reply:
> Let's still do _replace params.leverage with params['leverage']_ for it to work with all languages (without changing the transpiler specifically for this purpose).

`params.leverage` is proper jsdoc and it's needed to auto create docs, it's already transpiled in PHP, we just need the python transpilation